src/averageAges.js

```diff
@@ -55,7 +86,22 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child => {
+    return people.some(mother => {
+      if (onlyWithSon) {
+        return child.sex === 'm' && child.mother === mother.name;
+      } else {
+        return child.mother === mother.name;
+      }
+    });
+  });
```

Comment:
Let's make this code shorter, use ternary operator here and return value after arrow (without `return` keyword)